core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/AuditLog.java

```diff
@@ -169,6 +179,11 @@ public AuditLog build() {
             if (this.recordedAt == null) {
                 recordedAt = Instant.now().toString();
             }
+            String impersonator = MDC.get(IMPERSONATOR);
+            if (this.impersonatorId == null && StringUtils.isNotEmpty(impersonator)) {
+
+                impersonatorId = impersonator;
+            }
```

Comment:
At glance this gives the idea that, we initialize `impersonatorId` with `impersonator` which may not be an id.. I guess its not the case, and you just wanted to two variables names.. If thats the case shall we update the logic as 

```
            String impersonatorIdInMDC = MDC.get(IMPERSONATOR);
            if (this.impersonatorId == null && StringUtils.isNotEmpty(impersonatorIdInMDC)) {

                this.impersonatorId = impersonatorIdInMDC;
            }
```

Or if we can guarantee that MDC don't accept empty values or setting empty a `impersonatorId` doesn't do a logical difference, we can keep the logic simple as it does for the `requestId`

```
            if (this.impersonatorId == null) {
                impersonatorId = MDC.get(IMPERSONATOR);
            }
```

Reply:
It's guaranteed that MDC.get(IMPERSONATOR) doesn't have empty string hence Ill go second option